flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -1245,6 +1258,21 @@ private ApplicationReport startAppMaster(
         return report;
     }
 
+    private void setTokensFor(ContainerLaunchContext containerLaunchContext) throws IOException {
+        LOG.info("Adding delegation tokens to the AM container.");
+
+        Credentials credentials = UserGroupInformation.getCurrentUser().getCredentials();
+
+        DelegationTokenManager delegationTokenManager =
+                new DelegationTokenManager(flinkConfiguration);
+        delegationTokenManager.obtainDelegationTokens(credentials);
```

Comment:
`obtainDelegationTokens` obtains tokens but these are mustn't be invalidated if you mean that.
`start/stop` just starts and stops automatic re-obtain and propagation. Not yet see why stopping this needs to be called close. In this case `close` would be no-op because manual obtain is used.

Reply:
No, UGI is used which doesn't need to be closed.